.github/CONTRIBUTING.md

```diff
@@ -69,102 +67,142 @@ There are few ways you can play with the build:
+### Troubleshooting test suites
 
-We run test suite on every commit and requires 100% test pass. If the test suite did not complete successfully, they are likely:
+We run test suites on every commit and requires 100% test pass. If test suites did not complete successfully, they are likely:
 
+-  New changes causing fail in existing tests
 -  Intermittent services issue
 -  Test reliability issue, please see [#2938](https://github.com/microsoft/BotFramework-WebChat/issues/2938)
 -  Polluted development environment, for example:
    -  Outdated `node_modules` content
    -  Outdated Node.js or NPM
 
-When the test suite failed:
+When the test suites failed:
+
+-  Identify if it is related to new changes or not
+   -  Clone a clean repository and run the test suites without any changes
+-  Identify if it is intermittent issue or not
+   -  Run the test suites again
+
+If existing test suites failed without any changes:
+
+-  Test suites always fail
+   1. Service could be updated and causing a break in our test suites, please [file an issue](https://github.com/microsoft/BotFramework-WebChat/issues/new/choose)
+-  Test suites fail intermittently
+   1. If related to service issue, such as the MockBot is down, please [file an issue](https://github.com/microsoft/BotFramework-WebChat/issues/new/choose)
+   1. If related to test reliability, please comment to [#2938](https://github.com/microsoft/BotFramework-WebChat/issues/2938)
 
-1. Re-run the test suite
-   1. If it succeeded but a specific test fails intermittently, please comment to [#2938](https://github.com/microsoft/BotFramework-WebChat/issues/2938)
-1. If it continues to fail, please fresh clone the repository and run the test suite without your changes
-   1. If you suspect your environment is polluted, please use [Windows Sandbox](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-sandbox/windows-sandbox-overview) and/or a new Linux distro on WSL to verify
-1. If it still fails repeatedly, please [file an issue](https://github.com/microsoft/BotFramework-WebChat/issues/new/choose) to us.
+To debug race conditions:
+
+1. Append `location.reload()` to the test page so it will keep reloading the test page until it fail
+1. Navigate to the test page manually and wait until it failed and stop reloading
+
+General tips on race conditions or intermittent test failures:
+
+-  After sending a message, wait for responses from bot
+-  After a long message is shown, wait for scroll to complete
+-  Remove or speed up animations and media progress bars
 
 ## Static code analysis
 
-Run `npm run eslint` for static code analysis.
+Run `npm run precommit` for static code analysis.
 
-If you need to skip any ESLint errors, we recommend `eslint-disable-next-line`. Comments are required on why the rule is disabled.
+To skip any ESLint errors, we prefer `eslint-disable-next-line` instead of disabling a specific rule for the whole file. Comments are required on why the rule is disabled.
```

Comment:
```suggestion
To skip any ESLint errors, please use `eslint-disable-next-line` instead of disabling a specific rule for the whole file. Comments are required on why the rule is disabled.
```